scripts/templates/build_fatpart

```diff
@@ -257,6 +257,16 @@ function add_tree_cmd {
    link_or_copy $tc/$arch/tree_cmd/tree usr/bin
 }
 
+function add_lua {
```

Comment:
I don't know what happened but my long comment here disappeared. Maybe I didn't save it?
So, I'll re-write it, from scratch :-(

With the `add_lua` function written like that, `lua` will be always built in Tilck's image if it's present in the toolchain. At the beginning I used this approach too, but then the amount of apps to be loaded became too big and it's not convenient to remove/re-add an app from the toolchain every time. Therefore, I added a CMake variable for each package. See `add_vim`. In other words, here we need something like:

```
   if [ -z "@EXTRA_LUA_ENABLED@" ]; then
      return
   fi
```

Of course, we have to define that variable as well.
The place for that is `other/cmake/extra_apps.cmake`.
Just add:
```
if (EXISTS ${TCROOT}/${ARCH}/lua)
   set(EXTRA_LUA OFF CACHE BOOL "Load LUA in Tilck")
   message(STATUS "EXTRA_LUA: ${EXTRA_LUA}")
endif()
```
Below `if (EXISTS ${TCROOT}/${ARCH}/tree_cmd)` and then, at the bottom of the file,
add:

```
if (EXTRA_LUA)
   set(EXTRA_LUA_ENABLED "1")
endif()
```

Finally, we have to add an entry like:
```
lua = @EXTRA_LUA@
```

In `other/cmake/config_fatpart` in order to trigger the rebuild of the `fatpart` when the `EXTRA_LUA` option is changed.

Note: the most convenient way I have to change on-the-fly the cached CMake options is to run:
```
ccmake ./build
```

`ccmake` is a simple text-based UI for reconfiguring CMake projects.
Of course, you can use ./scripts/cmake_run -DEXTRA_LUA=1 but with `ccmake` you see all the configuration options at once.

I believe that's all :-)

Reply:
Hey @vvaltchev 
I have modified `other/cmake/extra_apps.cmake`, `other/cmake/config_fatpart` and `scripts/templates/build_fatpart` to include the changes you mentioned. Also I have tested these changes.